commit 33e719f5a38b548b98d6c5cbc49fa502fadd1a0e (refs/changes/80/3580/4)
Author: Jason King <jason.king@joyent.com>
Date:   2018-03-12T15:08:55-05:00 (1 year, 7 months ago)
    
    OS-6647 smartos build dying in ncurses with syntax error
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/ncurses/Makefile b/ncurses/Makefile
index 00510c7..e7fd906 100644
--- a/ncurses/Makefile
+++ b/ncurses/Makefile
@@ -18,7 +18,7 @@
 #
 # CDDL HEADER END
 #
-# Copyright (c) 2012, Joyent, Inc.  All rights reserved.
+# Copyright (c) 2018, Joyent, Inc.  All rights reserved.
 #
 
 VER =	ncurses-5.7
@@ -36,6 +36,12 @@ AUTOCONF_OPTS += \
 AUTOCONF_OPTS.64 += \
 	--libdir=/usr/gnu/lib/amd64
 
+# /bin/sh (aka ksh93)'s cat builtin likes to omit the final newline, which
+# causes problems with generating headers.sh using successive here docs.
+# Use bash for now to avoid lines being merged by this.
+AUTOCONF_ENV += 	CONFIG_SHELL=/bin/bash
+AUTOCONF_ENV.64 +=	CONFIG_SHELL=/bin/bash
+
 LDFLAGS +=	-R/usr/gnu/lib
 LDFLAGS.64 +=	-R/usr/gnu/lib/amd64
 
